Realm/RLMRealmUtil.mm

```diff
@@ -209,13 +209,16 @@ - (instancetype)initWithRealm:(RLMRealm *)realm error:(NSError **)error {
         _shutdownReadFd = pipeFd[0];
         _shutdownWriteFd = pipeFd[1];
 
-        [NSThread detachNewThreadSelector:@selector(listen) toTarget:self withObject:nil];
+        NSThread *thread = [[NSThread alloc] initWithTarget:self selector:@selector(listen) object:nil];
+        // Use the minimum allowed stack size, as we need very little in our listener
```

Comment:
I had to dig around a bit to confirm that 16KB is indeed the minimum allowed value, so maybe include a link to https://developer.apple.com/library/ios/documentation/Cocoa/Conceptual/Multithreading/CreatingThreads/CreatingThreads.html#//apple_ref/doc/uid/10000057i-CH15-SW7 in your comment (or more appropriate source if you have one).